### PR TITLE
Contextual roles corner cases

### PR DESCRIPTION
**Pull request by Jym77**
Closes https://github.com/web-platform-tests/interop-accessibility/issues/174

This add some corner cases for elements whose role depend on the presence of a name, by adding degenerate cases of "no name" (such as empty `title`, `aria-labelledby` to non-existing element, …)

The `<form>` test are in a `tentative` file because I am not sure what the expected role really is (they should still be treated as "form", but without being exposed as "landmark").

I haven't added cases here for `<div role="region">`.
It mostly behaves the same way as the new cases here (with browsers giving different results), but this is from [the Core AAM](https://www.w3.org/TR/core-aam-1.2/#role-map-region-nameless); the `core-aam` directory contains only manual tests that look very different from the `html-aam` ones, so I'm not sure where the `<div role="region">` tests would fit best (is it OK to add an automated test file in `core-aam` directory?)

### CODE REVIEW

File: html-aam/roles-contextual.tentative.html [hunk 1]
```diff
@@ -28,8 +28,33 @@
...
+<form data-testname="el-form-aria-label" aria-label="x" data-expectedrole="form" class="ex">x</form>
+<form data-testname="el-form-aria-label-empty" aria-label="" class="ex-generic">x</form>
+<form data-testname="el-form-aria-label-whitespace" aria-label=" " class="ex-generic">x</form>
+<form data-testname="el-form-aria-labelledby" aria-labelledby="labelledby" data-expectedrole="form" class="ex">x</form>
+<form data-testname="el-form-aria-labelledby-non-existing" aria-labelledby="non-existing" class="ex-generic">x</form>
+<form data-testname="el-form-aria-labelledby-empty" aria-labelledby="empty" class="ex-generic">x</form>
+<form data-testname="el-form-aria-labelledby-whitespace" aria-labelledby="space" class="ex-generic">x</form>
+<form data-testname="el-form-title" title="x" data-expectedrole="form" class="ex">x</form>
+<form data-testname="el-form-title-empty" title="" class="ex-generic">x</form>
+<form data-testname="el-form-title-whitespace" title=" " class="ex-generic">x</form>
```
Comment (spectranaut) [inline html-aam/roles-contextual.tentative.html:47]: I actually think the spec is clear in this case -- I think all of these should be "form": HTML-AAM refers to Core-AAM, and in Core-AAM the [computed role for form with an accessible name is "form"](https://w3c.github.io/core-aam/#role-map-form) and the [computed role for form without an accessible name is "form"](https://w3c.github.io/core-aam/#role-map-form-nameless). The computed name is different than how the element is exposed in accessibility APIs, so whether or not these elements are exposed as a landmark cannot be tested by testing the internal role.

**Review by spectranaut — CHANGES_REQUESTED** [review]
Thanks for doing this, overall great tests! See my comment about forms, however.

As for `role=region` tests, no one has made automated Core-AAM tests yet, feel free to add tests to that directory :)

**Comment by Jym77** [comment]
Thanks. I've removed the `form` tests and added some `<div role="region">` ones.

**Review by spectranaut — CHANGES_REQUESTED** [review]
Great! Thanks for the quick turn around, the tests look good. One minor thing before merging -- the new file you created has CR endings: https://github.com/web-platform-tests/wpt/pull/50679/checks?check_run_id=38300461526 

Search the lint tool rules here for how to fix: https://web-platform-tests.org/writing-tests/lint-tool.html

You can set git to switch the file endings to LF before committing -- google for how to do this with your set up?

**Review by spectranaut — APPROVED** [review]

**Comment by sideshowbarker** [comment]
@Jym77 Are the expectations for the `el-img-empty-alt-title` subtest here at odds with the expectations for the https://wpt.live/accname/name/comp_tooltip.tentative.html test? And with the HTML-AAM spec requirements?

Specifically, at https://w3c.github.io/html-aam/#img-element-accessible-name-computation the spec requires `alt` to be used _“even if its value is the empty string”_ — and requires `title` to be used only _“if there is no alt attribute”._

But isn’t the `el-img-empty-alt-title` expecting to opposite of that? That is, isn’t it expecting the `alt` attribute to be ignored, and the `title` attribute to be used?

See https://github.com/w3c/aria/pull/2378#issuecomment-2521438359 for related discussion — and for background on when the https://wpt.live/accname/name/comp_tooltip.tentative.html test got updated.

But that said, see also https://github.com/web-platform-tests/wpt/pull/49552#issuecomment-2559671371 — where I point out that the current current requirements in the HTML-AAM spec seem to be at odds with the requirements in the accname spec.

**Comment by sideshowbarker** [comment]
After looking at the results in the major engines for the tests added in this PR, and after looking at the spec, I think all the -labelledby-whitespace and -labelledby-empty tests added in this PR should be removed. So I’ve opened https://github.com/web-platform-tests/wpt/pull/51604 for that.

For the record here: The reasons for removing those tests are:

- If you take a look at https://wpt.fyi/results/html-aam/roles-contextual.html, you’ll see that those are failing in at least two major engines — and some are failing in all three major engines.
- It’s not clear if the test expectations in those test match any existing requirements that are actually defined in any spec.

To put it in other terms: If two (or more) different implementors working from the spec requirements did not produce implementations which pass a particular test, then it seems there might be a high likelihood that the spec doesn’t actually clearly require the behavior the test is testing.

**Comment by cookiecrook** [comment]
Agreed. And if they are required in a spec, they should be in a tentative file, not the main one.
